driver/src/test/scala/BulkOpsSpec.scala

```diff
@@ -79,6 +77,34 @@ class BulkOpsSpec(implicit ee: ExecutionEnv)
         }
       }
     }
+
+    s"take minimal size into account into total size" in {
+      bulks[BSONDocument](
+        documents = producer2Docs,
+        maxBsonSize = doc1.byteSize,
+        maxBulkSize = 2)(_.byteSize) must beLike[BulkProducer[BSONDocument]] {
+        case prod1 =>
+          prod1() must beLeft
+      }
+    }
+
+    s"take into account the fact that keys increase in size in a larger array" in {
+      val ExpectedFirstBulk = Iterator.continually(doc1).take(10).toList
+      val ExpectedSecondBulk = List(doc1)
+      val enoughSizeWithoutConsideringSizeIncrease = (doc1.byteSize + 1 + BSONElementSet.typePrefixByteSize) * 11
```

Comment:
Let keep it in `maxBsonSize` (the value is used once)

Reply:
Will do. This was to make it clear why I choose this max size.